### PR TITLE
Flex reorder strategy

### PR DESCRIPTION
**Pull request by gbalint**
Flex reordering strategy.

No special controls or any visual effects, animation etc., it just the implements the flex reordering itself

Limitations:
- I expect the elements appear in the flex order in the screen. If they have manual offsets then this may not be true.
- It only supports single selection.

### CODE REVIEW

File: editor/src/components/canvas/canvas-strategies/flex-reorder-strategy.tsx [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+      canvasPoint({
+        x: (frames[i].x + frames[i].width + frames[i + 1].x) / 2,
+        y: (frames[i].y + frames[i].height + frames[i + 1].y) / 2,
+      }),
+    )
+  }
+  return [firstPoint, ...middlePoints, lastPoint]
+}
+
+function reverseIndexIfRightToLeft(idx: number, length: number, isLeftToRight: boolean) {
```
Comment (balazsbajorics) [inline editor/src/components/canvas/canvas-strategies/flex-reorder-strategy.tsx:158]: nice

**Review by balazsbajorics — APPROVED** [review]

File: editor/src/components/canvas/canvas-strategies/flex-reorder-strategy.tsx [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+    ) {
+      return []
+    }
+    const { selectedElements } = canvasState
+    const target = selectedElements[0]
+
+    const siblingsOfTarget = MetadataUtils.getSiblings(sessionState.startingMetadata, target).map(
+      (element) => element.elementPath,
+    )
+    const flexSiblingsOfTarget = siblingsOfTarget.filter((child) =>
```
Comment (seanparsons) [inline editor/src/components/canvas/canvas-strategies/flex-reorder-strategy.tsx:54]: Might be worth having a test for the case where one of the siblings returns false for this conditional.

**Review by seanparsons — APPROVED** [review]

File: editor/src/components/canvas/canvas-strategies/flex-reorder-strategy.tsx (hunk 2 above)
Comment (gbalint) [reply editor/src/components/canvas/canvas-strategies/flex-reorder-strategy.tsx:54]: Such a good point! I wrote it and realized that case has a bug! Fixing it now

Comment (gbalint) [reply editor/src/components/canvas/canvas-strategies/flex-reorder-strategy.tsx]: Fixed in 05f01159ddffc52b3a9d01283d7d027560861899

**Pull request merged**